### PR TITLE
Refactoring, bug fixes, and performance improvements

### PR DESCRIPTION
**Pull request by kabiroberai**
TODO before merging:
- [x] Switch back spaces to tabs
- [x] Add better ways to dismiss the keyboard

### CODE REVIEW

**Comment by kabiroberai** [comment]
Worth mentioning that this closes #1

File: Alderis/ColorPickerSlidersViewController.swift [hunk 1]
```diff
@@ -141,139 +152,49 @@ class ColorPickerSlidersViewController: UIViewController, ColorPickerTabProtocol
...
 			topSpacerView.heightAnchor.constraint(equalToConstant: 0),
 			hexTextField.widthAnchor.constraint(equalToConstant: 100),
-			hexStackView.heightAnchor.constraint(equalTo: anyColorRow.heightAnchor)
+			hexStackView.heightAnchor.constraint(equalTo: colorRows[0].heightAnchor)
 		])
 
 		updateMode()
 	}
 
-	override func viewDidLayoutSubviews() {
```
Comment (kabiroberai) [inline Alderis/ColorPickerSlidersViewController.swift]: @kirb is there a reason that `updateColor` was being called in `viewDidLayoutSubviews`? Removing it resolves #2 but I hope that doesn't lead to a regression.

**Comment by ThatsNiceGuy** [comment]
not to hurry anyone or anything but I was just wondering what happened to merging this PR

**Comment by kabiroberai** [comment]
@ThatsNiceGuy from what I can tell, this has already been merged by rebasing my commits. @kirb lmk whether I should close this PR if you’re done merging it

**Comment by kirb** [comment]
It’s definitely been fully merged (rebased + squashed), so we can close this.

**Comment by ThatsNiceGuy** [comment]
again not to be rude, but does anyone know if this is gonna be pushed to the repo? as of today the version on chariz does not have this functionality

**Pull request closed without merging**